src/components/viewmodels/rooms/RoomSummaryCardViewModel.tsx

```diff
@@ -0,0 +1,290 @@
+    };
+
+    const onRoomExtensionsClick = (): void => {
+        RightPanelStore.instance.pushCard({ phase: RightPanelPhases.Extensions }, true);
+    };
+
+    const onRoomPinsClick = (): void => {
+        PosthogTrackers.trackInteraction("PinnedMessageRoomInfoButton");
+        RightPanelStore.instance.pushCard({ phase: RightPanelPhases.PinnedMessages }, true);
+    };
+
+    const onRoomSettingsClick = (ev: Event): void => {
+        defaultDispatcher.dispatch({ action: "open_room_settings" });
+        PosthogTrackers.trackInteraction("WebRightPanelRoomInfoSettingsButton", ev);
+    };
+
+    const onShareRoomClick = (): void => {
+        Modal.createDialog(ShareDialog, {
+            target: room,
+        });
+    };
+
+    const onRoomExportClick = async (): Promise<void> => {
+        Modal.createDialog(ExportDialog, {
+            room,
+        });
+    };
+
+    const onRoomPollHistoryClick = (): void => {
+        Modal.createDialog(PollHistoryDialog, {
+            room,
+            matrixClient: cli,
+            permalinkCreator,
+        });
+    };
+
+    const onLeaveRoomClick = (): void => {
+        defaultDispatcher.dispatch({
+            action: "leave_room",
+            room_id: room.roomId,
+        });
+    };
+
+    const onReportRoomClick = async (): Promise<void> => {
+        const [leave] = await Modal.createDialog(ReportRoomDialog, {
+            roomId: room.roomId,
+        }).finished;
+        if (leave) {
+            defaultDispatcher.dispatch({
+                action: "leave_room",
+                room_id: room.roomId,
+            });
+        }
+    };
+
+    const onFavoriteToggleClick = (): void => {
+        tagRoom(room, DefaultTagID.Favourite);
+    };
+
+    const onInviteToRoomClick = (): void => {
+        inviteToRoom(room);
+    };
+
+    // Room Search element ref
+    const searchInputRef = useRef<HTMLInputElement>(null);
+
+    // Onkeydown press escape event, clear the search field
+    const onUpdateSearchInput = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+        if (searchInputRef.current && e.key === Key.ESCAPE) {
+            searchInputRef.current.value = "";
+            onSearchCancel?.();
+        }
+    };
+
+    // Focus the search field when the user clicks on the search button component
+    useDispatcher(defaultDispatcher, (payload) => {
+        if (payload.action === Action.FocusMessageSearch) {
+            searchInputRef.current?.focus();
+        }
+    });
+
+    // Clear the search field when the user leaves the search view
+    useTransition(
+        (prevTimelineRenderingType) => {
+            if (
+                prevTimelineRenderingType === TimelineRenderingType.Search &&
+                roomContext.timelineRenderingType !== TimelineRenderingType.Search &&
+                searchInputRef.current
+            ) {
+                searchInputRef.current.value = "";
+            }
+        },
+        [roomContext.timelineRenderingType],
+    );
```

Comment:
We can move the code related to the search input to its own custom hook in the file and use it here.